# Generated by using Rcpp::compileAttributes() -> do not edit by hand
# Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

#' PMR Individual Rcpp Helper
#'
#' Individual helper function which does the work in C++, returns it to R. Call
#' PMR_individual to use the function.
#'
#' @param yin standardized exposure vector (e.g. gene expression in TWAS).
#' @param zin standardized complex trait vector.
#' @param x1in standardized cis-genotype matrix in eQTL data.
#' @param x2in standardized cis-genotype matrix in GWAS data.
#' @param gammain indicator variable for constrained PMR model, with 1 for the null 
#' model that there is no horiozntal pleiotropy.
#' @param alphain indicator variable for constrained PMR model, with 1 for the null 
#' model that there is no causal effect.
#' @param max_iterin The maximum iteration.
#' @param epsin The convergence tolerance of the absolute value of the difference 
#' between the nth and (n+1)th loglikelihood.
#' 
#' @return a list of estimates of model parameters, including the causal effect 
#' \code{alpha}, the horizontal pleiotropy effect \code{gamma}, and the two 
#' corresponding p values
#' 
PMR_individual_rcpp <- function(yin, zin, x1in, x2in, gammain, alphain, max_iterin, epsin) {
    .Call(`_PMR_PMR_individual`, yin, zin, x1in, x2in, gammain, alphain, max_iterin, epsin)
}

#' PMR Summary Rcpp Helper
#'
#' Summary helper function which does the work in C++, returns it to R. Call
#' PMR_summary to use the function.
#'
#' @param betaxin the cis-SNP effect size vector for one specific gene in eQTL 
#' data, which must be calculated based on both the standardized gene expression 
#' value and the standardized cis-genotype matrix.
#' @param betayin the cis-SNP effect size vector for one specific gene in GWAS 
#' data, which be calculated based on both the standardized complex trait value 
#' and the standardized cis-genotype matrix.
#' @param Sigma1sin the LD matrix in eQTL data.
#' @param Sigma2sin the LD matrix in GWAS data.Both \code{Sigma2sin} and 
#' \code{sigma1sin} are often the same from the reference panel.
#' @param samplen1 the sample size of eQTL data.
#' @param samplen2 the sample size of GWAS data.
#' @param gammain indicator variable for constrained model, with 1 for the null 
#' model that there is no horiozntal pleiotropy.
#' @param alphain indicator variable for constrained model, with 1 for the null 
#' model that there is no causal effect.
#' @param max_iterin The maximum iteration.
#' @param epsin The convergence tolerance of the absolute value of the difference 
#' between the nth and (n+1)th loglikelihood.
#' 
#' @return A list of estimates of model parameters, including the causal effect 
#' \code{alpha}, the horizontal pleiotropy effect \code{gamma}, and the two corresponding p values.
#' 
PMR_summary_rcpp <- function(betaxin, betayin, Sigma1sin, Sigma2sin, samplen1, samplen2, gammain, alphain, max_iterin, epsin) {
    .Call(`_PMR_PMR_summary`, betaxin, betayin, Sigma1sin, Sigma2sin, samplen1, samplen2, gammain, alphain, max_iterin, epsin)
}

